google/cloud/bigtable/data/_metrics/handlers/opentelemetry.py

```diff
@@ -0,0 +1,196 @@
+            description="A count of the number of attempts that failed to reach Google's network.",
+        )
+        self.application_latencies = meter.create_histogram(
+            name="application_latencies",
+            description="A distribution of the total latency introduced by your application when Cloud Bigtable has available response data but your application has not consumed it.",
+            unit="ms",
+        )
+        self.throttling_latencies = meter.create_histogram(
+            name="throttling_latencies",
+            description="The latency introduced by the client by blocking on sending more requests to the server when there are too many pending requests in bulk operations.",
+            unit="ms",
+        )
+
+
+class OpenTelemetryMetricsHandler(MetricsHandler):
+    """
+    Maintains a set of OpenTelemetry metrics for the Bigtable client library,
+    and updates them with each completed operation and attempt.
+
+    The OpenTelemetry metrics that are tracked are as follows:
+      - operation_latencies: latency of each client method call, over all of it's attempts.
+      - first_response_latencies: latency of receiving the first row in a ReadRows operation.
+      - attempt_latencies: latency of each client attempt RPC.
+      - retry_count: Number of additional RPCs sent after the initial attempt.
+      - server_latencies: latency recorded on the server side for each attempt.
+      - connectivity_error_count: number of attempts that failed to reach Google's network.
+      - application_latencies: the time spent waiting for the application to process the next response.
+      - throttling_latencies: latency introduced by waiting when there are too many outstanding requests in a bulk operation.
+    """
+
+    def __init__(
+        self,
+        *,
+        project_id: str,
+        instance_id: str,
+        table_id: str,
+        app_profile_id: str | None = None,
+        client_uid: str | None = None,
+        instruments: _OpenTelemetryInstruments = _OpenTelemetryInstruments(),
+        **kwargs,
+    ):
+        super().__init__()
+        self.otel = instruments
+        # fixed labels sent with each metric update
+        self.shared_labels = {
+            "client_name": f"python-bigtable/{bigtable_version}",
+            "client_uid": client_uid or str(uuid4()),
+            "resource_project": project_id,
+            "resource_instance": instance_id,
+            "resource_table": table_id,
+        }
+        if app_profile_id:
+            self.shared_labels["app_profile"] = app_profile_id
+
+    def on_operation_complete(self, op: CompletedOperationMetric) -> None:
+        """
+        Update the metrics associated with a completed operation:
+          - operation_latencies
+          - retry_count
+        """
+        try:
+            status = str(op.final_status.value[0])
+        except (IndexError, TypeError):
+            status = "2"  # unknown
```

Comment:
we actually export status string instead of the numeric value, so OK, DEADLINE_EXCEEDED, etc.